### PR TITLE
chore: adding mode scripts

### PR DESCRIPTION
**Pull request by kupermind**
- Adding Mode scripts for deployment.

### CODE REVIEW

**Review by 77ph — APPROVED** [review]

File: docs/configuration.json [hunk 1]
```diff
@@ -647,5 +647,92 @@
...
+      },
+      {
+        "name": "GnosisSafeMultisig",
+        "artifact": "abis/0.8.28/GnosisSafeMultisig.json",
+        "address": "0x0a50009D55Ed5700ac8FF713709d5Ad5fa843896"
+      },
+      {
+        "name": "GnosisSafeSameAddressMultisig",
+        "artifact": "abis/0.8.28/GnosisSafeSameAddressMultisig.json",
+        "address": "0x77290FF625fc576f465D0256F6a12Ce4480a5b8a"
```
Comment (DavidMinarsch) [inline docs/configuration.json:734]: where did we take these from?

File: scripts/deployment/l2/globals_mode_mainnet.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{"contractVerification":true,"useLedger":true,"derivationPath":"m/44'/60'/2'/0/0","providerName":"base","networkURL":"https://mainnet.mode.network","gasPriceInGwei":"1","gnosisSafeAddress":"0x69f4D1788e39c87893C980c06EdF4b7f686e2938","gnosisSafeProxyFactoryAddress":"0xC22834581EbC8527d974F8a1c97E1bEA4EF910BC","baseURI":"https://gateway.autonolas.tech/ipfs/","serviceRegistryName":"Service Registry L2","serviceRegistrySymbol":"AUTONOLAS-SERVICE-L2-V1","bridgeMediatorAddress":"","olasAddress":"","multisigProxyHash130":"0xb89c1b3bdf2cf8827818646bce9a8f6e372885f8c55e5c07acbd307cb133b000","serviceRegistryAddress":"","operatorWhitelistAddress":"","serviceRegistryTokenUtilityAddress":"","serviceManagerTokenAddress":"","gnosisSafeMultisigImplementationAddress":"","gnosisSafeSameAddressMultisigImplementationAddress":"","stakingTokenAddress":"","stakingNativeTokenAddress":"","minStakingDepositLimit":"10000000000000000000000","timeForEmissionsLimit":"2592000","numServicesLimit":"100","apyLimit":"3000000000000000000","stakingVerifierAddress":"","stakingFactoryAddress":"","L1CrossDomainMessengerProxyAddress":"0x95bDCA6c8EdEB69C98Bd5bd17660BaCef1298A6f"}
```
Comment (DavidMinarsch) [inline scripts/deployment/l2/globals_mode_mainnet.json:1]: where did we take the gnosis addresses from?

**Review by DavidMinarsch — DISMISSED** [review]

File: scripts/deployment/l2/globals_mode_mainnet.json (hunk 2 above)
Comment (kupermind) [reply scripts/deployment/l2/globals_mode_mainnet.json:1]: https://github.com/safe-global/safe-deployments/blob/main/src/assets/v1.3.0/gnosis_safe.json

**Review by 77ph — APPROVED** [review]

File: scripts/deployment/l2/globals_mode_mainnet.json (hunk 2 above)
Comment (kupermind) [reply scripts/deployment/l2/globals_mode_mainnet.json]: Let's use the canonical one as historically we use the first Safe in the list of addresses. Technically there is no contract difference, it's the matter of which factory was used to deploy Safe contracts.

**Review by 77ph — APPROVED** [review]

**Pull request merged**